.github/workflows/go.yml

```diff
@@ -0,0 +1,69 @@
+name: Go
+
+on:
+  push:
+    branches:
+      - main
+      - release/*
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+  pull_request:
+
+jobs:
+  build:
+    name: Build
+
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version:
+          - 1.16.x
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go release
+        uses: percona-platform/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Checkout code
+        uses: percona-platform/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Prepare - Init
+        run: make init
+
+      - name: Prepare - Format
+        run: make format
+
+      - name: Check differences
+        run: |
+          git status
+          git diff --exit-code
+
+      - name: Run reviewdog
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: make ci-reviewdog
+
+      - name: Install
+        run: make install
+
+      - name: Run Tests
+        run: |
+          make test-cover
+          make test-crosscover
+          make test-race
+
+      - name: Run required checkers and linters.
+        run: make check
+
+      - name: Run debug commands on failure
+        if: ${{ failure() }}
+        run: |
+          env
+          go version
+          go env
+          pwd
+          git status
```

Comment:
how this can help identify why it is failing? I think reporting from tests should be sufficient